src/basic/log.h

```diff
@@ -367,3 +368,23 @@ int log_syntax_invalid_utf8_internal(
 #define DEBUG_LOGGING _unlikely_(log_get_max_level() >= LOG_DEBUG)
 
 void log_setup(void);
+
+#define log_every_x_usec(usec, level, ...)              \
+        ({                                              \
+                static RateLimit rl = {                 \
+                        .interval = (usec),             \
+                        .burst = 1                      \
+                };                                      \
+                if (ratelimit_below(&rl))               \
+                        log_full(level, __VA_ARGS__);   \
+        })
```

Comment:
We don't need statement expressions here, so you can drop the parentheses around the block.